t/apicast-oidc.t

```diff
@@ -362,3 +362,43 @@ my $jwt = 'eyJraWQiOiJzb21la2lkIiwiYWxnIjoiSFMyNTYifQ.'.
 "Authorization: Bearer $jwt"
 --- error_log
 [jwt] alg mismatch
+
+=== TEST 8: Key not available
+(Happens when the token belongs to a different realm)
```

Comment:
I think the more accurate test description is that the "token was signed by a different key". The same key could be used accross multiple realms but with this fix the `iss` claim will be correctly validated against the `issuer` field in the proxy config. Equally if the `kid`s don't match then it won't even find the jwk.

In fact we can see that further down in this test, we have different realms but the logic never gets that far in the validation process because it fails before that with _"[jwk] jwk not found"_

Reply:
added a test for the verification of the expected issuer